BEST_PRACTICES.md

```diff
@@ -0,0 +1,146 @@
+  - Methods within this class handle satellite operations, especially for reconfiguring the satellite via installation processes etc. Other operational methods are part of the Satellite mixins, including factories. e.g `sat.capsule_certs_generate()`
+  - This class inherits methods from ContentHost and Capsule classes, hence methods from these classes are available with this class objects.
+- **Capsule**:
+  - This class represents the Capsule host and includes essential properties and methods that describe the Capsule host.
+  - Properties like `url` and `is_stream` specify details about the specific Capsule host.
+  - The `satellite` property returns the satellite object to which the Capsule is connected.
+  - Methods within this class manage operations related to the Capsule, especially for reconfiguring it via installation processes. Other operational methods are part of the Capsule mixins. e.g `sat.restart_services()`
+  - This class inherits methods from ContentHost class, hence methods from these classes are available with this class objects.
+- **ContentHost**:
+  - This class represents the ContentHost or Client, including crucial properties and methods that describe these hosts.
+  - Properties like `hostname` specify details about specific ContentHosts.
+  - The `nailgun_host` property returns the API/nailgun object for the ContentHost.
+  - Methods within this class manage operations on ContentHosts, particularly for reconfiguring them. Other operational methods are part of the ContentHost mixins.
+  - This class inherits methods from `Brokers` modules `Host` class, hence methods from that class are available with this class objects.
+
+By adhering to these best practices, you'll contribute to the efficient organization and accessibility of helpers within the framework, enhancing the overall development experience and maintainability
+ of the project. Thank you for your commitment to best practices!
+
+
+## FAQs:
+
+_**Q1. When should I prefer/not prefer fixtures over to helper function when implementing a new helper ?**_
+
+_Answer:_ The answers could be multiple. Fixture over functions are preferred when:
+   - A new helper function should be cached based on scope of the function usage.
+   - A new helper has dependency on other fixtures.
+   - A new helper provides the facility of setup and teardown in the helper function itself.
+   - A new helper as a fixture accept parameters, allowing you to create dynamic test scenarios.
+
+_**Q2. Where should I implement the fixture that could be used by all tests from all endpoints?**_
+  _Answer:_ The global package `pytest_plugins` is the recommended place. Choose the right subpackage `core` if framework fixture `component` if component fixture.
+
+_**Q3. Where should we implement the helper functions needed by fixtures ?**_
+
+  _Answer:_ It is prefered to use utils but if the function is not reusable one can keep in the fixture module and access locally.
```

Comment:
I'd suggest that, if the helper function is only usable by the fixture, then it should live in the fixture's own module. However, if it could be used elsewhere in robottelo/tests then it should become a util.